ddtrace/tracer/textmap.go

```diff
@@ -1428,23 +1428,20 @@ func (*propagatorBaggage) extractTextMap(reader TextMapReader) (*SpanContext, er
 		return &ctx, nil
 	}
 
-	keyVals := strings.Split(baggageHeader, ",")
-	for _, kv := range keyVals {
-		// Split on the first instance of "=" i.e, `a=b=c` becomes `a: b=c`.
-		key, val, ok := strings.Cut(kv, "=")
-		if !ok {
-			log.Warn("invalid baggage item: %s, dropping", kv)
-		}
-		key = strings.TrimSpace(key)
-		val = strings.TrimSpace(val)
-		if key == "" || val == "" {
-			log.Warn("invalid baggage item: '%s', dropping", kv)
-		}
-		key, errKey := url.QueryUnescape(key)
-		val, errVal := url.QueryUnescape(val)
-		if errKey != nil || errVal != nil {
-			log.Warn("failed to decode baggage item: %s, dropping", kv)
+	// 1) validation pass
+	parts := strings.Split(baggageHeader, ",")
+	for _, kv := range parts {
+		k, v, ok := strings.Cut(kv, "=")
+		if !ok || strings.TrimSpace(k) == "" || strings.TrimSpace(v) == "" {
+			log.Warn("invalid baggage item: %q, dropping entire header", kv)
+			return &ctx, nil
 		}
```

Comment:
I can get behind this two-pass approach, but I do see an opportunity to do a small optimization to avoid doing strings.TrimSpace on the same k/v twice:
You could change line 1435 to store variables for `strings.TrimSpace(k)` and `strings.TrimSpace(v)`, then introduce an else block after line 1438 that basically sets k and v to those respective values.